include/openssl/ssl.h

```diff
@@ -1849,16 +1849,18 @@ OPENSSL_EXPORT int SSL_get_extms_support(const SSL *ssl);
 // not been negotiated yet.
 OPENSSL_EXPORT const SSL_CIPHER *SSL_get_current_cipher(const SSL *ssl);
 
-// SSL_get_client_ciphers returns stack of ciphers offered by the client during
-// a handshake. If the |ssl| is a client or the handshake hasn't occurred yet,
-// NULL is returned. The stack of ciphers IS NOT de/serialized, so NULL will
-// also be returned for deserialized or transported |ssl|'s that haven't yet
+// SSL_get_client_ciphers returns the stack of ciphers offered by the client during
+// a handshake and that are supported by this library. If the |ssl| is a client or the
```

Comment:
Is it the ones supported by AWS-LC, or the potentially matching ciphers with what the server accepts?

Reply:
No cipher choosing is occurring yet. SSL_get_client_ciphers returns client_cipher_suites which is the parsed variant of all_client_cipher_suites (added in this PR). This parsed variant contains all cipher suites offered from the client that are supported by our library (checking against server supported suites doesn't occur yet). client_cipher_suites is then used to pick a cipher that is supported both by the server and client.